### ISSUE TITLE
tdssrv: login.c does not support connections from current Microsoft products

### ISSUE THREAD

**Issue by smorey2** [FreeTDS/freetds#340]
freddy77, et al,

canceling my previous pull request, and creating issues so we can discuss first.

.net core *.Data.SqlClient, and Microsoft SQL Management Studio (SSMS) both fail when attempting to connect to a listening freetds server port.

tds_alloc_read_login() receives a TDS71_PRELOGIN, but assigns tds_version = 0x701, only 0x702 >= are supported these products anymore.

then .net connects but SSMS fails when product_version is not set correctly.

might be helpful for others like me, to have a way of providing this information, or it commented somewhere, took me a long time to figure this out.

also 0x702 adds a query_head, so the generic_query is incorrect.

SSMS also requires a response to its 'capabilities' query, before completing its connection.

**Comment by freddy77** [comment]
I was not aware MS tools bump the minimum requirement to 7.2. Not an issue. About SSMS I won't be that worried, unless you plan to support a lot of tables/databases/store procedures to make it happy it won't work that much. The status of the server part is really basic, far way from other part. The only part updated is basically the part needed for implementing the pool server. About `TDS71_PRELOGIN` setting only version 7.1 don't worry, it will be bumped automatically when login is received.

As said the status of server part is really basic, feel free to change. For the header consider that all packets need header for 7.2, not only `TDS_QUERY`.

Not sure what you intend by the `capabilities` query.

**Comment by smorey2** [comment]
SSMS Query tool does not need all the tables/databases/store proc calls, just an initial call. The SSMS studio does however, and I do plan on supporting some if not all.

Basic support would be just using a standard .net client library, to connect to my tds server.

I understand the server part is really basic, which is the part I am interested in, Perhaps I can help flush it out more, I noticed a lot of missing parts, primarily in query.c:tds_get_generic_query() method. I would like to discuss this more at some point if you had time, perhaps an argument callback for each type.

TDS71_PRELOGIN setting will not be bumped automatically, because the server.c:tds_send_login_ack() rejects the connection because the tds_version of 7.1 is sent.

I have simplified my "pull-request", adding comments to the header, and the query_head method, no interface changes: https://github.com/netcode-io/freetds/commit/5d718f6f5a08cc0511bcf89470746c7245eecfc7

I noticed there are two branches, master and active1.1, are pull requests made to both?

Sorry, the `capabilities` query was the SSMS Query tool initial one, asking some version information before the connection is finally accepted.

One again, thank you for your time freddy77.
